src/leo_gateway_s3_api.erl

```diff
@@ -508,8 +532,142 @@ handle_1(Req, [{NumOfMinLayers, NumOfMaxLayers}, HasInnerCache, CustomHeaderSett
+                                                                              chunk_sign= <<>>},
+                                  put(io_list, <<>>),
+                                  put(dec_state, wait_size),
+                                  put(chunk_offset, 0),
+                                  put(sign_params, AWSChunkSignParams),
+                                  AwsChunkedFun = fun(Bin) ->
+                                                          IOList    = get(io_list),
+                                                          DecState  = get(dec_state),
+                                                          Offset = get(chunk_offset),
+                                                          SignParams = get(sign_params),
+%                                                          ?debug("aws_chunk_decode", "IOList Size: ~p, DecState: ~p, ChunkSize: ~p", [iolist_size(IOList), DecState, ChunkSize]),
+%                                                          ?debug("aws_chunk_decode", "SignParams: ~p", [SignParams]),
+                                                          Ret = aws_chunk_decode({ok, <<>>}, [IOList, Bin], DecState, Offset, SignParams),
+                                                          case Ret of
+                                                              {{error, Reason2}, {_, _, _, _}} ->
+%                                                                  ?error("aws_chunk_decode", "Parse Error: ~p", [Reason2]),
+                                                                  {error, Reason2};
+                                                              {{ok, Acc}, {IOList_2, DecState_2, Offset_2, SignParams_2}} ->
+                                                                  put(io_list, IOList_2),
+                                                                  put(dec_state, DecState_2),
+                                                                  put(chunk_offset, Offset_2),
+                                                                  put(sign_params, SignParams_2),
+                                                                  OutBin = iolist_to_binary(Acc),
+%                                                                  ?debug("aws_chunk_decode", "Parse Complete: ~p", [byte_size(OutBin)]),
+                                                                  erlang:garbage_collect(self()),
+                                                                  {ok, OutBin}
+                                                          end
+                                                  end,
+                                  ReqParams#req_params{content_decode_fun = AwsChunkedFun};
+                              _ ->
+                                  ReqParams
+                          end;
+                      _ ->
+                          ReqParams
+                  end,
+
```

Comment:
yes. 
All APIs exposed to users are supposed to be called inside handle/2 callback, 
so leo_gateway require handling exceptions if needed,
otherwise cowboy catch those exceptions at here.

https://github.com/ninenines/cowboy/blob/1.0.x/src/cowboy_handler.erl#L111-L126